wandb/sdk_py27/wandb_artifacts.py

```diff
@@ -225,56 +204,53 @@ def add_reference(self, uri, name=None, checksum=True, max_objects=None):
 
         return manifest_entries
 
-    # TODO: name this add_obj?
     def add(self, obj, name):
         """adds `obj` to the artifact, located at `name`. You can use Artifact#get(`name`) after downloading
         the artifact to retrieve this object.
         
         Arguments:
-        - `obj`:wandb.Media - the object to save
+        - `obj`:wandb.WBValue - the object to save
         - `name`:str - the path to save
         """
-        if isinstance(obj, Media):
 
-            # If the object is coming from another artifact, save it as a reference
-            if hasattr(obj, "_source") and obj._source is not None:
-                suffix = "." + obj.get_json_suffix() + ".json"
-                ref_path = obj._source["artifact"].get_path(
-                    obj._source["name"] + suffix
-                )
-                path = name + suffix
-                return self.add_reference(ref_path, path)[0]
-
-            # Otherwise, save the object directly via json
-            elif type(obj) in JSONABLE_MEDIA_CLASSES:
-                obj_id = id(obj)
-                if obj_id in self._added_objs:
-                    return self._added_objs[obj_id]
-                val = obj.to_json(self)
-                if "_type" not in val:
-                    val["_type"] = obj.get_json_suffix()
-                suffix = val["_type"] + ".json"
-                if not name.endswith(suffix):
-                    name = name + "." + suffix
-                entry = self._manifest.get_entry_by_path(name)
-                if entry is not None:
-                    return entry
-                with self.new_file(name) as f:
-                    import json
-
-                    # TODO: Do we need to open with utf-8 codec?
-                    f.write(json.dumps(obj.to_json(self), sort_keys=True))
-                # Note, we add the file from our temp directory.
-                # It will be added again later on finalize, but succeed since
-                # the checksum should match
-                entry = self.add_file(os.path.join(self._artifact_dir.name, name), name)
-                self._added_objs[obj_id] = entry
-                return entry
-            else:
-                ValueError("Can't add obj of type {} to artifact".format(type(obj)))
-        else:
+        # Validate that the object is wandb.Media type
+        if not isinstance(obj, WBValue):
             raise ValueError("Can't add obj to artifact")
```

Comment:
Should this say something to the effect of `only WBValue objects can be added`?

Reply:
Changed